rust/agama-server/src/network/web.rs

```diff
@@ -0,0 +1,242 @@
+))]
+async fn general_state(State(state): State<NetworkState>) -> Json<GeneralState> {
+    let (tx, rx) = oneshot::channel();
+    state.actions.send(Action::GetGeneralState(tx)).unwrap();
+
+    let state = rx.await.unwrap();
+
+    Json(state)
+}
+
+#[utoipa::path(post, path = "/network/state", responses(
+  (status = 200, description = "Update general network config", body = GenereralState)
+))]
+async fn update_general_state(
+    State(state): State<NetworkState>,
+    Json(value): Json<GeneralState>,
+) -> Result<Json<GeneralState>, NetworkError> {
+    state
+        .actions
+        .send(Action::UpdateGeneralState(value.clone()))
+        .unwrap();
+
+    let (tx, rx) = oneshot::channel();
+    state.actions.send(Action::GetGeneralState(tx)).unwrap();
+    let state = rx.await.unwrap();
+
+    Ok(Json(state))
+}
+
+#[utoipa::path(get, path = "/network/wifi", responses(
+  (status = 200, description = "List of wireless networks", body = Vec<String>)
+))]
+async fn wifi_networks(State(state): State<NetworkState>) -> Json<Vec<String>> {
+    let (tx, rx) = oneshot::channel();
+    state.actions.send(Action::RefreshScan(tx)).unwrap();
+    let _ = rx.await.unwrap();
+    let (tx, rx) = oneshot::channel();
+    state.actions.send(Action::GetAccessPoints(tx)).unwrap();
+
+    let access_points = rx.await.unwrap();
+    let mut networks = vec![];
+    for ap in access_points {
+        let ssid = ap.ssid.to_string();
+        if !ssid.is_empty() && !networks.contains(&ssid) {
+            networks.push(ssid);
+        }
+    }
+
+    Json(networks)
+}
+
+#[utoipa::path(get, path = "/network/devices", responses(
+  (status = 200, description = "List of devices", body = Vec<Device>)
+))]
+async fn devices(State(state): State<NetworkState>) -> Json<Vec<Device>> {
+    let (tx, rx) = oneshot::channel();
+    state.actions.send(Action::GetDevices(tx)).unwrap();
+
+    Json(rx.await.unwrap())
+}
+
+#[utoipa::path(get, path = "/network/connections", responses(
+  (status = 200, description = "List of known connections", body = Vec<NetworkConnection>)
+))]
+async fn connections(State(state): State<NetworkState>) -> Json<Vec<NetworkConnection>> {
+    let (tx, rx) = oneshot::channel();
+    state.actions.send(Action::GetConnections(tx)).unwrap();
+    let connections = rx.await.unwrap();
+    let connections = connections
+        .iter()
+        .map(|c| NetworkConnection::try_from(c.clone()).unwrap())
+        .collect();
+
+    Json(connections)
+}
+
+#[utoipa::path(post, path = "/network/connections", responses(
+  (status = 200, description = "Add a new connection", body = Connection)
+))]
+async fn add_connection(
+    State(state): State<NetworkState>,
+    Json(value): Json<NetworkConnection>,
+) -> Result<Json<Uuid>, NetworkError> {
+    let (tx, rx) = oneshot::channel();
+    let conn = NetworkConnection::from(value);
+
+    state
+        .actions
+        .send(Action::AddConnection(
```

Comment:
IMHO we should remove the need of having separate actions (`AddConnection` and `UpdateConnection`).